pkg/clustermanager/eksa_mover_test.go

```diff
@@ -50,81 +51,76 @@ func newMoverTest(tb testing.TB) *moverTest {
 	}
 }
 
-func (a *moverTest) buildClient(err1, err2 error, objs ...kubernetes.Object) {
-	a.fromClient = test.NewFakeKubeClient(clientutil.ObjectsToClientObjects(objs)...)
-	a.toClient = test.NewFakeKubeClient()
-	a.clientFactory.EXPECT().BuildClientFromKubeconfig(a.bootstrap.KubeconfigFile).Return(a.fromClient, err1).MaxTimes(1)
-	a.clientFactory.EXPECT().BuildClientFromKubeconfig(a.mgmtCluster.KubeconfigFile).Return(a.toClient, err2).MaxTimes(1)
-}
-
-func (a *moverTest) expectClientError() {
-	a.fromClient = test.NewFakeKubeClientAlwaysError()
-	a.clientFactory.EXPECT().BuildClientFromKubeconfig(a.bootstrap.KubeconfigFile).Return(a.fromClient, nil)
-	a.toClient = test.NewFakeKubeClientAlwaysError()
-	a.clientFactory.EXPECT().BuildClientFromKubeconfig(a.mgmtCluster.KubeconfigFile).Return(a.toClient, nil)
+func (a *moverTest) buildClients(fromObjs, toObjs []kubernetes.Object) {
+	a.fromClient = test.NewFakeKubeClient(clientutil.ObjectsToClientObjects(fromObjs)...)
+	a.toClient = test.NewFakeKubeClient(clientutil.ObjectsToClientObjects(toObjs)...)
 }
 
 func TestMoverSuccess(t *testing.T) {
 	tt := newMoverTest(t)
 	objs := tt.spec.ClusterAndChildren()
-	tt.buildClient(nil, nil, objs...)
+	tt.buildClients(objs, nil)
 	m := clustermanager.NewMover(tt.log, tt.clientFactory,
 		clustermanager.WithMoverRetryBackOff(time.Millisecond),
 		clustermanager.WithMoverNoTimeouts(),
 	)
 
-	tt.Expect(m.Move(tt.ctx, tt.spec, tt.bootstrap, tt.mgmtCluster)).To(gomega.Succeed())
+	tt.Expect(m.Move(tt.ctx, tt.spec, tt.fromClient, tt.toClient)).To(gomega.Succeed())
 
 	for _, obj := range objs {
-		tt.Expect(tt.toClient.Get(tt.ctx, obj.GetName(), obj.GetNamespace(), obj)).To(gomega.Succeed())
+		u := &unstructured.Unstructured{}
+		u.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+		tt.Expect(tt.toClient.Get(tt.ctx, obj.GetName(), obj.GetNamespace(), u)).To(gomega.Succeed())
+		original, err := runtime.DefaultUnstructuredConverter.ToUnstructured(obj)
+		tt.Expect(err).To(gomega.Succeed())
+		tt.Expect(u.Object["spec"]).To(gomega.Equal(original["spec"]))
```

Comment:
nit: this will give you a better diff error if the test fails
```suggestion
		tt.Expect(u.Object["spec"]).To(gomega.BeComparable(original["spec"]))
```